### PR TITLE
[ArkHomes] Update to latest version of API, added perms & points

### PR DESCRIPTION
**Pull request by paperrain**
Updated with latest api.
Added permissions check
Added ArkShop points
Fix plugins.unload

### CODE REVIEW

**Comment by Lethalinjectionx** [comment]
Please add "IsPluginLoaded()" checks for ArkShop and Permission code area to prevent crashes if either of these plugins are not being used by the server. Basically it should ignore all code related to them if the plugins are not loaded.

ArkApi::Tools::IsPluginLoaded("Permissions")
ArkApi::Tools::IsPluginLoaded("ArkShop")

**Comment by Lethalinjectionx** [comment]
Includes and Libraries should be added in a way that allows for the for the API and Plugin repos to be installed next to each other.

For example:
c:\ARK-Server-API\
c:\Ark-Server-Plugins\

This makes the relative paths for all plugins the same in the public repo making it easier to download and compile everything.

<AdditionalIncludeDirectories>..\Includes;..\..\..\ARK-Server-API\version\Core\Public;..\..\Permissions\Permissions\Public;..\..\ArkShop\ArkShop\Public;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>

<AdditionalLibraryDirectories>..\..\..\ARK-Server-API\out_lib;..\..\ArkShop\out_lib;..\..\Permissions\out_lib;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>

**Comment by paperrain** [comment]
Done, thank you for your guidance!

File: ArkHomes/ArkHomes/ArkHomes.vcxproj [hunk 1]
```diff
@@ -46,14 +46,15 @@
...
       <LanguageStandard>stdcpp17</LanguageStandard>
-      <AdditionalIncludeDirectories>..\Includes;\..\..\ARK-Server-API\version\Core\Public;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>..\Includes;..\..\..\ARK-Server-API\version\Core\Public;..\..\Permissions\Permissions\Public;..\..\ArkShop\ArkShop\Public;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
     </ClCompile>
     <Link>
       <SubSystem>Windows</SubSystem>
       <EnableCOMDATFolding>true</EnableCOMDATFolding>
       <OptimizeReferences>true</OptimizeReferences>
       <GenerateDebugInformation>true</GenerateDebugInformation>
       <AdditionalLibraryDirectories>\..\..\ARK-Server-API\out_lib;\..\Ark-Server-Plugins\Permissions\out_lib;\..\ArkShop\out_lib;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
Comment (Lethalinjectionx) [inline ArkHomes/ArkHomes/ArkHomes.vcxproj:56]: Looks like there is an old and new entry for "AdditionalLibraryDirectories"

Comment (Lethalinjectionx) [inline ArkHomes/ArkHomes/ArkHomes.vcxproj:56]: The "AdditionalLibraryDirectories" is listed twice now with the old and new updates.

Comment (Lethalinjectionx) [reply ArkHomes/ArkHomes/ArkHomes.vcxproj]: Ok, thanks.

**Pull request merged**